packages/devkit/src/utils/package-json.ts

```diff
@@ -448,10 +453,16 @@ export function ensurePackage<T extends any = any>(
   const tempDir = dirSync().name;
 
   console.log(`Fetching ${pkg}...`);
-  execSync(`${getPackageManagerCommand().addDev} ${pkg}@${requiredVersion}`, {
-    cwd: tempDir,
-    stdio: 'ignore',
-  });
+  const isVerbose = process.env.NX_VERBOSE_LOGGING === 'true';
+  execSync(
+    `${
+      getPackageManagerCommand(detectPackageManager(), tempDir).addDev
```

Comment:
In the case of `pnpm` workspace, it would try to run the `pnpm add -Dw ...` command, which fails since `tempDir` is not a workspace.